src/caches.jl

```
@@ -119,18 +119,15 @@ function OrdinaryDiffEq.alg_cache(
     SolProj = solution_space_projection(FAC, is_secondorder_ode)
 
     # Prior dynamics
-    prior = if alg.prior isa IWP
-        IWP{uElType}(d, alg.prior.num_derivatives)
-    elseif alg.prior isa IOUP && ismissing(alg.prior.rate_parameter)
-        r = Array{uElType}(calloc, d, d)
-        IOUP{uElType}(d, q, r, alg.prior.update_rate_parameter)
-    elseif alg.prior isa IOUP
-        IOUP{uElType}(d, q, alg.prior.rate_parameter, alg.prior.update_rate_parameter)
-    elseif alg.prior isa Matern
-        Matern{uElType}(d, q, alg.prior.lengthscale)
-    else
-        error("Invalid prior $(alg.prior)")
+    @assert (dim(alg.prior) == 1 || dim(alg.prior) == d)
+    prior = remake(alg.prior; elType=uElType, dim=d)
+    if (
+        (prior isa IOUP) && (prior.update_rate_parameter) &&
+        (prior.rate_parameter isa Missing)
```

This should probably also throw a better error and be tested.